dashboard/app/controllers/policy_compliance_controller.rb

```diff
@@ -0,0 +1,86 @@
+class PolicyComplianceController < ApplicationController
+  before_action :authenticate_user!, except: [:child_account_consent]
+
+  # GET /policy_compliance/child_account_consent
+  # URL where parents or guardians give consent for their child to have an
+  # account on Code.org
+  def child_account_consent
+    @permission_granted = false
+    token = request.params[:token]
+    return render status: :bad_request if token.blank?
+    #Get parent permission request
+    permission_request = ParentalPermissionRequest.find_by(uuid: token)
+    return render status: :bad_request if permission_request.nil?
+    #Get User
+    user = permission_request.user
+    #Update User
+    if user.child_account_compliance_state != User::ChildAccountCompliance::PERMISSION_GRANTED
+      user.update_child_account_compliance(User::ChildAccountCompliance::PERMISSION_GRANTED)
+      user.save!
+      parent_email = permission_request.parent_email
+      ParentMailer.parent_permission_confirmation(parent_email).deliver_now
+    end
+    @permission_granted = true
+    @permission_granted_date = user.child_account_compliance_state_last_updated
+  end
+
+  # POST /policy_compliance/child_account_consent
+  # This URL creates a parental permission request. It is invoked via the
+  # lockout panel or potentially user settings by the child account.
+  def child_account_consent_request
+    # If we already comply, don't suddenly invalid it
+    if current_user.child_account_compliance_state == User::ChildAccountCompliance::PERMISSION_GRANTED
+      redirect_back fallback_location: '/lockout' and return
+    end
+
+    # Only allow three unique permission requests per day
+    date = Date.today
+    permission_requests = ParentalPermissionRequest.where(
+      created_at: date.midnight..date.end_of_day
+    ).limit(3).count
+
+    # Create a ParentalPermissionRequest token for user and parent email
+    # When the student 'updates' the parental email, we actually just create a
+    # new request row.
+    permission_request = ParentalPermissionRequest.find_or_create_by(
+      user: current_user,
+      parent_email: params[:"parent-email"]
+    )
+
+    # If we are making a new request but already sent too many today,
+    # just bail and return whence we came
+    if !permission_request.persisted? && permission_requests >= 3
```

Comment:
`permission_request.persisted?` check doesn't have much sense because `ParentalPermissionRequest.find_or_create_by` will find or create a new record which means it is always `persisted`

Reply:
I'll change to find or initialize. good catch.

Reply:
```suggestion
    if permission_request.new_record? && permission_requests >= 3
```

Reply:
yes. exactly this. find_or_initialize and then new_record. i want to avoid persisting the record if the limit for the day is met.